packages/dx-react-grid-demos/src/demo-sources/grid-featured-chart-integration/material-ui/demo.jsx

```diff
@@ -0,0 +1,195 @@
+  TableRowDetail,
+} from '@devexpress/dx-react-grid-material-ui';
+import { withStyles } from '@material-ui/core/styles';
+import { citiesCount, regionsCount } from '../../../demo-data/chart-data';
+
+const detailContainerStyles = theme => ({
+  detailContainer: {
+    marginBottom: 3 * theme.spacing.unit,
+  },
+  title: {
+    color: theme.palette.text.primary,
+    fontSize: theme.typography.fontSize,
+  },
+  paper: {
+    paddingTop: 3.5 * theme.spacing.unit,
+  },
+});
+const legendStyles = () => ({
+  root: {
+    display: 'flex',
+    margin: 'auto',
+    flexDirection: 'row',
+  },
+});
+const legendLabelStyles = () => ({
+  label: {
+    whiteSpace: 'nowrap',
+  },
+});
+
+const nullComponent = () => null;
+const currencyFormatter = ({ value }) => `$${value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
+const axisLabel = ({
+  text,
+  ...restProps
+}) => <ValueAxis.Label text={currencyFormatter({ value: text })} {...restProps} />;
+
+const CurrencyTypeProvider = props => (
+  <DataTypeProvider
+    {...props}
+    formatterComponent={currencyFormatter}
+  />
+);
+
+const legendRootBase = ({ classes, ...restProps }) => (
+  <Legend.Root
+    {...restProps}
+    className={classes.root}
+  />
+);
+const legendRoot = withStyles(legendStyles, { name: 'LegendRoot' })(legendRootBase);
+
+const legendLabelBase = ({ classes, ...restProps }) => (
+  <Legend.Label className={classes.label} {...restProps} />
+);
+const legendLabel = withStyles(legendLabelStyles, { name: 'LegendLabel' })(legendLabelBase);
+
+const BarSeriesForCity = regionCities => Object
+  .keys(regionCities[0])
+  .reduce((acc, item, index) => {
+    if (item !== 'year') {
+      acc.push(
+        <BarSeries
+          key={index.toString()}
+          valueField={item}
+          argumentField="year"
+          name={item}
+        />,
+      );
+    }
+    return acc;
+  }, []);
+
+const gridDetailContainerBase = data => ({ row, classes }) => {
+  const regionCities = data.reduce((acc, item) => {
+    const currentCities = item.cities.reduce((current, itemCity) => {
+      let currentObj = {};
+      if (itemCity.region === row.region) {
+        currentObj = { [itemCity.cityName]: itemCity.count };
+      }
+      return { ...current, ...currentObj };
+    }, []);
+    return [...acc, { year: item.year, ...currentCities }];
+  }, []);
+
+  return (
+    <div className={classes.detailContainer}>
+      <h5 className={classes.title}>
+        {`Economics of ${row.region}`}
+      </h5>
+      <Paper className={classes.paper}>
+        <Chart
+          data={regionCities}
+          height={300}
+        >
+          <ArgumentAxis
+            name="year"
+            type="band"
+            tickComponent={nullComponent}
+          />
+          <ValueAxis
+            labelComponent={axisLabel}
+            tickComponent={nullComponent}
+            lineComponent={nullComponent}
+          />
+          <ChartGrid />
+          {BarSeriesForCity(regionCities)}
```

Comment:
Can we use `BarSeriesForCity` as a React component?

```
<BarSeriesForCity regionCities={regionCities} />
```

Otherwise, uppercase looks strange.